Create tests for `users` query

pedro-m-dietrich opened the pull request:
The tests verify if the `users` query for default values for `userLimit` and `offset`, for specified values and if unauthenticated users fail the request.

pedro-m-dietrich committed:
Create tests for `users` query

The tests verify if the `users` query for default values for `userLimit`
and `offset`, for specified values and if unauthenticated users fail the
request.

alanraso commented on test/server.ts:
```diff
@@ -329,4 +329,142 @@ describe('User API', function () {
+        users: {
+          users: expectedList,
+          totalUsers: dbUsers.length,
+          userCount: expectedList.length,
```
Olhando agora, acredito que esse campo não seja necesário porque ele pode ser trivialmente inferido no tamanho do array. Sempre bom evitar reundância.

alanraso commented on test/server.ts:
```diff
@@ -329,4 +329,142 @@ describe('User API', function () {
+      });
+    });
+
+    it('should retrieve an user list with specified values', async function () {
```
Acho que seria interessante um caso de teste para um cenário em que o offset é maior do que o total de usuários. O que acha?

pedro-m-dietrich replied:
Adicionado.

pedro-m-dietrich committed:
Remove `userCount` field from `users` response

The `userCount` has been removed, as this value can be easily obtained
from the size of the `users` array.

A test for an offset greater than the amount of users in the database
has been added.

matheusflds commented on test/server.ts:
```diff
@@ -329,4 +329,161 @@ describe('User API', function () {
     });
   });
+
+  describe('#Users query', async function () {
```
acho que esse arquivo ta ficando meio grande, dps pode ser uma boa ver de quebrar em um pra cada describe 🤔

pedro-m-dietrich replied:
Concordo.
Vou dar uma refatorada no código e nos testes em um novo PR.
